pkg/client/client.go

```diff
@@ -83,7 +84,10 @@ func (c *Client) Version() (version.ServiceClient, error) {
 	if err := c.checkConn(); err != nil {
 		return nil, err
 	}
-	return version.NewServiceClient(c.conn), nil
+	if c.versionServiceClient == nil {
```

Comment:
Is this something strictly necessary? If yes, why? :)

Reply:
@leodido I wasn't able to come up with a way to inject the mock at test time without checking that `c.versionServiceClient` is nil. For example, when calling `c.Version()` in the test, without this check, `c.versionServiceClient` would always be overridden and set to the real versionServiceClient, rather than the mock. This way, the tests can set the `c.versionServiceClient` to the mock versionServiceClient, and this `if` can skip over setting it to the real thing. 

Thanks for the review!

Reply:
Hi @leodido @ndhanushkodi
Any update on that? It would be great to have this merged!

Reply:
Hi, sorry for the delay. I should be able to have this done by the end of the weekend!